pkg/cache/clusterqueue_snapshot.go

```diff
@@ -112,35 +114,44 @@ func (c *ClusterQueueSnapshot) PotentialAvailable(fr resources.FlavorResource) i
 	return potentialAvailable(c, fr)
 }
 
-// The methods below implement several interfaces. See
-// dominantResourceShareNode, resourceGroupNode, and netQuotaNode.
+func (c *ClusterQueueSnapshot) GetName() string {
+	return c.Name
+}
+
+// Implements dominantResourceShareNode interface.
 
 func (c *ClusterQueueSnapshot) fairWeight() *resource.Quantity {
 	return &c.FairWeight
 }
 
-func (c *ClusterQueueSnapshot) usageFor(fr resources.FlavorResource) int64 {
-	return c.ResourceNode.Usage[fr]
-}
+// The methods below implement hierarchicalResourceNode interface.
 
-func (c *ClusterQueueSnapshot) resourceGroups() []ResourceGroup {
-	return c.ResourceGroups
+func (c *ClusterQueueSnapshot) getResourceNode() ResourceNode {
+	return c.ResourceNode
 }
 
-func (c *ClusterQueueSnapshot) parentResources() ResourceNode {
-	return c.Parent().ResourceNode
+func (c *ClusterQueueSnapshot) parentHRN() hierarchicalResourceNode {
+	return c.Parent()
 }
 
-func (c *ClusterQueueSnapshot) GetName() string {
-	return c.Name
+// DominantResourceShare returns a value from 0 to 1,000,000 representing the maximum of the ratios
+// of usage above nominal quota to the lendable resources in the cohort, among all the resources
+// provided by the ClusterQueue, and divided by the weight.
+// If zero, it means that the usage of the ClusterQueue is below the nominal quota.
+// The function also returns the resource name that yielded this value.
+// Also for a weight of zero, this will return 9223372036854775807.
```

Comment:
Should this description be moved to `dominantResourceShare`?

Reply:
Done.